### PR TITLE
added clone's solution hw-5.1

### PR DESCRIPTION
**Pull request by Serhii-Maksymov**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,36 @@
...
-public class Car {
+public class Car implements Cloneable {
+    private String modelCare;
+    private int carSpeed;
+    private String carColor;
+    private int weight;
+    private Engine engine;
 
+    public Car() {
+    }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java:11]: why did you create this constructor?

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        Car newCar = (Car) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java:31]: please read the checklist before sending the task for review

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
-public class Engine {
+public class Engine implements Cloneable {
+    private String engineType;
+    private int cylindersNumber;
+    private String transmissionType;
+    private int enginePower;
+    private int engineTorque;
 
+    public Engine() {
+    }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java:11]: do you need it?

File: src/main/java/core/basesyntax/Engine.java [hunk 4]
```diff
@@ -1,5 +1,26 @@
...
+        this.engineType = engineType;
+        this.cylindersNumber = cylindersNumber;
+        this.transmissionType = transmissionType;
+        this.enginePower = enginePower;
+        this.engineTorque = engineTorque;
+    }
+
+    @Override
+    protected Engine clone() throws CloneNotSupportedException {
+        return (Engine) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java:24]: is it correct overriding?

File: src/main/java/core/basesyntax/Engine.java (hunk 3 above)
Comment (Serhii-Maksymov) [reply src/main/java/core/basesyntax/Engine.java]: No, I don't need it.

File: src/main/java/core/basesyntax/Car.java (hunk 2 above)
Comment (Serhii-Maksymov) [reply src/main/java/core/basesyntax/Car.java]: Sorry, I will be attentive.

File: src/main/java/core/basesyntax/Engine.java (hunk 4 above)
Comment (Serhii-Maksymov) [reply src/main/java/core/basesyntax/Engine.java]: apparently not

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
+    private int weight;
+    private Engine engine;
 
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() {
+        Car newCar = new Car();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java:16]: set all fields with constructor

File: src/main/java/core/basesyntax/Engine.java [hunk 6]
```diff
@@ -1,5 +1,14 @@
...
+public class Engine implements Cloneable {
+    private String engineType;
+    private int cylindersNumber;
+    private String transmissionType;
+    private int enginePower;
+    private int engineTorque;
 
+    @Override
+    protected Engine clone() {
+        return new Engine();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java:12]: how about fields?

File: src/main/java/core/basesyntax/Engine.java [hunk 7]
```diff
@@ -1,5 +1,34 @@
...
+        this.enginePower = enginePower;
+    }
+
+    public void setEngineTorque(int engineTorque) {
+        this.engineTorque = engineTorque;
+    }
+
+    @Override
+    protected Engine clone() {
+        return new Engine();
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:32]: will we get the same engine with the same variable's values?

**Comment by Tarabarnik** [comment]
my last comments are not fix or answered

**Pull request closed without merging**